src/arch/cuda/hybridAPI/hapi_functions.h

```diff
@@ -43,7 +43,7 @@ AMPI_CUSTOM_FUNC(cudaError_t, hapiFreeHostPool, void*)
 AMPI_CUSTOM_FUNC(cudaError_t, hapiMemcpyAsync, void*, const void*, size_t, enum cudaMemcpyKind, cudaStream_t)
 
 // Explicit memory allocations using pinned memory pool.
-AMPI_CUSTOM_FUNC(void*, hapiPoolMalloc, size_t)
+AMPI_CUSTOM_FUNC(void, hapiPoolMalloc, void**, size_t)
```

Comment:
`cudaMallocHost` returns `cudaError_t`. Should `hapiPoolMalloc` do the same? Same question for `hapiPoolFree`.

Reply:
Good point, I think they should for `hapiCheck` to work (which checks the `cudaError_t` return value).